rs/state_layout/src/state_layout.rs

```diff
@@ -2425,8 +2547,10 @@ where
 
 /// Copies the given file and ensures that the `read/write` permission of the
 /// target file match the given permission.
+/// This function is used for files inside checkpoints, so the `src` file is intended to be
+/// marked readonly.
 /// Syncs the target file if `fsync` is true.
-fn copy_file_and_set_permissions(
+fn copy_checkpoint_file(
```

Comment:
Just realized from  PR #702   that `copy_file_and_set_permissions` will be also be used in some cases without FSync at the end.
Is this why we have a separate `copy_checkpoint_file()`?